public/app/features/dashboard/state/settings/actions.ts

```diff
@@ -0,0 +1,31 @@
+import { ThunkResult } from '../../../../types';
+import { setVariableSettings } from './reducer';
+import { DashboardVariablesSettings } from './types';
+import { DashboardModel } from '../DashboardModel';
+import { reportInteraction } from '../../../../../../packages/grafana-runtime';
+
+export function initDashboardSettings(dashboard: DashboardModel): ThunkResult<void> {
+  return function (dispatch) {
+    const variableSettings: DashboardVariablesSettings | undefined = dashboard.getVariableSettings();
+    if (variableSettings) {
+      dispatch(setVariableSettings(variableSettings));
+      reportInteraction('Dashboard variable settings loaded', { showUnknowns: Number(variableSettings.showUnknowns) });
```

Comment:
I'm curious about `variableSettings.showUnknowns` getting cast to a number 🤔

Reply:
Hehe, I'm guessing this gets pushed into some time series DB and I thought that `booleans` weren't the best for that.

Reply:
Ah, fair enough!